### PR TITLE
Disable escape in single quote

### LINKED ISSUE #22: Disallow escape sequence in single quote 'p'
asdasdsadsadsa123

As shown in the pic, escaping in `'` isn't highlighted but in `"`.
`q` is equal to `'` so escaping shouldn't be highlighted.
`qq` is equal to `"` and works.

```perl
my $foo = q(
\n
Hello \x50
And hello \x{2603}\x{2602}
);

my $bar = qq(
\n
Hello \x50
And hello \x{2603}\x{2602}
);

my $baz = '\r\n';
my $awesomeBaz = "\r\n";
```

### PR DESCRIPTION
**Pull request by Fank**
This fixes #22 
quote_highlighting

### CODE REVIEW

**Comment by 50Wliu** [comment]
This isn't strictly related to your changes, but as far as I know, the Atom team loves to have tests flow naturally, eg `it "does not escape characters in single-quote strings"` instead of `it "'text'"`.

**Comment by Fank** [comment]
Thanks for the info ill update it

**Comment by Fank** [comment]
@50Wliu This is now more naturally or?

File: spec/grammar-spec.coffee [hunk 1]
```diff
@@ -42,31 +42,26 @@ describe "perl grammar", ->
       expect(tokens[6]).toEqual value: "gxr", scopes: ["source.perl", "string.regexp.replace.perl", "punctuation.definition.string.perl", "keyword.control.regexp-option.perl"]
 
   describe "tokenizes constant variables", ->
-    it "test __FILE__", ->
+    it "highlight constants", ->
```
Comment (50Wliu) [inline spec/grammar-spec.coffee]: highlights

File: spec/grammar-spec.coffee [hunk 2]
```diff
@@ -42,31 +42,26 @@ describe "perl grammar", ->
...
-    it "test __END__", ->
       {tokens} = grammar.tokenizeLine("__END__")
       expect(tokens[0]).toEqual value: "__END__", scopes: ["source.perl", "constant.language.perl"]
 
-    it "test __DATA__", ->
       {tokens} = grammar.tokenizeLine("__DATA__")
       expect(tokens[0]).toEqual value: "__DATA__", scopes: ["source.perl", "constant.language.perl"]
 
-    it "test non constant", ->
+    it "does not highlight custom constants like a default constant", ->
```
Comment (50Wliu) [inline spec/grammar-spec.coffee]: Is `like a default constant` needed?

File: spec/grammar-spec.coffee [hunk 3]
```diff
@@ -96,74 +91,102 @@ describe "perl grammar", ->
       expect(tokens[0]).toEqual value: "DESTROY", scopes: ["source.perl", "meta.function.perl", "entity.name.function.perl"]
 
   describe "tokenizes method calls", ->
-    it "test q (quoting) without brackets", ->
+    it "ignores highlight if called like a method", ->
```
Comment (50Wliu) [inline spec/grammar-spec.coffee]: `does not highlight`

File: spec/grammar-spec.coffee [hunk 4]
```diff
@@ -96,74 +91,102 @@ describe "perl grammar", ->
...
-      expect(lines[1][0]).toEqual value: "This is my first line", scopes: ["source.perl", "string.quoted.other.q.perl"]
-      expect(lines[2][0]).toEqual value: "and this the second one)", scopes: ["source.perl", "string.quoted.other.q.perl"]
+      expect(lines[1][0]).toEqual value: "This is my first line\\n", scopes: ["source.perl", "string.quoted.other.q.perl"]
+      expect(lines[2][0]).toEqual value: "and this the second one)\\x00", scopes: ["source.perl", "string.quoted.other.q.perl"]
       expect(lines[3][0]).toEqual value: "last", scopes: ["source.perl", "string.quoted.other.q.perl"]
       expect(lines[4][0]).toEqual value: "~", scopes: ["source.perl", "string.quoted.other.q.perl", "punctuation.definition.string.end.perl"]
       expect(lines[4][1]).toEqual value: ";", scopes: ["source.perl"]
 
+  describe "tokenizes double quoting", ->
+    it "\"text\"", ->
```
Comment (50Wliu) [inline spec/grammar-spec.coffee]: These three still haven't been converted.

**Comment by 50Wliu** [comment]
Other than some minor notes, great job!
`describe`s look like they might also need some work but I'm less clear on the rules for those.  Maybe @kevinsawicki can clarify?

**Comment by Fank** [comment]
@50Wliu Thanks for all the comments, i changed all `it`.

**Comment by kevinsawicki** [comment]
> @kevinsawicki can clarify?

Yeah, here is how example an example of the typical spec flow:

```coffee
describe "when tokenizing constant variables", ->
  it "creates tokens with the constant scope", ->
```

So basically when you combine the `describe` and the `it`, it flows like a natural sentence: "when tokenizing constant variables it creates tokens with the constant scope".  These kind of combined sentences are what is shown in the spec window or command line when they fail.

You might want to look at this core spec for more examples: https://github.com/atom/atom/blob/587ebd753840517dca899982d6a4f3931a1f1b39/spec/text-editor-spec.coffee#L21-L22

I'm okay merging this now though, thanks for fixing this.

**Pull request merged**